tensorflow/core/kernels/sparse_tensor_dense_matmul_op_gpu.cu.cc

```diff
@@ -71,13 +97,24 @@ __global__ void SparseTensorDenseMatMulKernel(
     // b_value == (ADJ_B) ? conj(b[j, k]) : b[k, j]
     const T b_input = ldg(b + ((ADJ_B) ? j * b_cols + k : k * b_cols + j));
     const T b_value = ADJ_B ? Eigen::numext::conj(b_input) : b_input;
-    GpuAtomicAdd(out_location,
-                 static_cast<Tsum>(a_value) * static_cast<Tsum>(b_value));
+
+    reduction<Tsum, T>(out_location, a_value * b_value);
   }
 }
 
 namespace functor {
 
+bool RequireDeterminism() {
+  static bool require_determinism = [] {
+    bool deterministic_ops = false;
+    TF_CHECK_OK(tensorflow::ReadBoolFromEnvVar("TF_DETERMINISTIC_OPS",
+                                               /*default_val=*/false,
+                                               &deterministic_ops));
+    return deterministic_ops;
+  }();
+  return require_determinism;
+}
+
```

Comment:
This function will need to be internally-linked, using an anonymous namespace ([like this](https://github.com/tensorflow/tensorflow/pull/47925/commits/dec160950c7531c9f86d5f8a0e7847fb91035b42)). Otherwise, it will probably collide with the other functions of the same name.

Later, I will factor the repetitive instances of this function into a shared library.

Reply:
Note to self: resolved.